src/Symfony/Component/Security/Core/User/UserInterface.php

```diff
@@ -69,13 +71,6 @@ public function getPassword();
      */
     public function getSalt();
 
-    /**
-     * Returns the username used to authenticate the user.
-     *
-     * @return string The username
-     */
-    public function getUsername();
```

Comment:
@wouterj @fabpot Removing a method from an interface isn't a BC-break ?

If I use `$user->getUsername()` in 5.2, I now can't be sure it still works in 5.3.
Also keeping the method would solve the psalm issues.

Reply:
See https://github.com/symfony/symfony/pull/41493